durablefunctionsmonitor-vscodeext/src/az-func-as-a-graph/traverseFunctionProject.ts

```diff
@@ -276,10 +278,36 @@ async function mapOrchestratorsAndActivitiesAsync(functions: FunctionsMap, proje
     if (isDotNet) {
         
         // Trying to extract extra binding info from C# code
-        for (const func of otherFunctions) {
+        for (const func of activities.concat(otherFunctions)) {
+
+            const bindingsFromFunctionJson = functions[func.name].bindings as { type: string, direction: string }[];
+            const bindingsFromCode = DotNetBindingsParser.tryExtractBindings(func.code);
+
+            const existingBindingTypes: string[] = bindingsFromFunctionJson.map(b => b.type);
+
+            for (let binding of bindingsFromCode) {
+
+                // Only pushing extracted binding, if a binding with that type doesn't exist yet in function.json,
+                // so that no duplicates are produced
+                if (!existingBindingTypes.includes(binding.type)) {
+                 
+                    bindingsFromFunctionJson.push(binding);
+                }
+            }
```

Comment:
What is the behavior before these changes were added? Were certain features not available for activity functions?

Reply:
Bindings could be shown duplicated.
In previous platform versions bindings were not written to function.json files - that's why we need DotNetBindingsParser.tryExtractBindings().
In recent versions they started to output all the bindings (yet not with full information, e.g. `direction` field is missing), so the output from DotNetBindingsParser.tryExtractBindings() collided with what's already in function.json.
This new code basically merges these two sources of binding info.
Also yes, it now shows binding for activities (those were not shown before).